scipy/optimize/tests/test_linprog.py

```diff
@@ -1348,51 +1331,75 @@ def test_network_flow(self):
 
 
 class TestLinprogSimplexDefault(LinprogSimplexTests):
-    options = {}
 
-    def test_bug_7237(self):
-        # we want to check for the error
+    def setup_method(self):
+        self.options = {}
+
+    def test_bug_5400(self):
+        with pytest.raises(ValueError):
+            super(TestLinprogSimplexDefault, self).test_bug_5400()
+
+    def test_bug_7237_low_tol(self):
+        # this will pass sucessfully with the new default tolerance of 1e-9
+        # however here we want to test the error is returned for tol=1e-12.
+        self.options.update({'tol': 1e-12})
         with pytest.raises(ValueError):
             super(TestLinprogSimplexDefault, self).test_bug_7237()
 
-    def test_bug_8174(self):
-        # we want to check for the warning about the pivot
+    @pytest.mark.xfail(reason='Fails with too low tolerance.')
```

Comment:
I don't understand why we're adding to the test _and_ marking it xfail?

Reply:
For Bland's rule it was meant to document that it is still failing. For the other two (Default, NoPresolve) it was supposed to check that it failed with a warning. Looking back at it I realise that's not what xfail does. I think the best way to change this would be to check for status 4 when tolerance is below some value.